climate-advisor/service/requirements.txt

```diff
@@ -1,10 +1,29 @@
-fastapi==0.110.0
-uvicorn[standard]==0.29.0
-httpx==0.27.0
-pydantic==2.7.1
-python-dotenv==1.0.1
-SQLAlchemy[asyncio]==2.0.30
-asyncpg==0.29.0
-alembic==1.13.1
-aiosqlite==0.19.0
-PyYAML==6.0.1
+# Core FastAPI and async dependencies (Latest: 0.111.x)
+fastapi>=0.104.0,<0.112.0
+uvicorn[standard]>=0.20.0,<0.31.0
+httpx>=0.24.0
+
+# Database dependencies (Latest: 2.0.30+)
+SQLAlchemy[asyncio]>=2.0.0,<3.0.0
+psycopg2-binary>=2.9.0
+asyncpg>=0.29.0
+alembic>=1.13.0
+
+# Configuration and utilities
+python-dotenv>=1.0.0
+PyYAML>=6.0.0
+aiosqlite>=0.19.0
+
+# Data model and validation (Latest: 2.8.x)
+pydantic>=2.5.0,<3.0.0
+
+# AI and embeddings (Latest: 1.35.x)
+openai>=1.0.0
+numpy>=1.24.0
+
+# Document processing (Latest: 3.0.1)
+PyPDF2>=3.0.0
+langchain-text-splitters>=0.2.0,<0.3.0
+
+# Vector database (Latest: 0.3.x)
+pgvector>=0.2.0
```

Comment:
Do we need these comments with latest version numbers here or can you try to upgrade all the dependencies to the latest version instead (maybe in a separate PR?)